core/src/main/java/org/apache/iceberg/SnapshotManager.java

```diff
@@ -54,6 +54,73 @@ public ManageSnapshots rollbackTo(long snapshotId) {
     return this;
   }
 
+  @Override
+  public ManageSnapshots createBranch(String name, long snapshotId) {
+    return createBranch(name, snapshotId, null, null, null);
+  }
+
+
+  @Override
+  public ManageSnapshots createBranch(
+      String name,
+      long snapshotId,
+      Long maxRefAgeMs,
+      Integer minSnapshotsToKeep,
+      Long maxSnapshotAgeMs) {
+
+    transaction
+        .manageSnapshotReferenceOperation()
+        .createBranch(name, snapshotId, maxRefAgeMs, minSnapshotsToKeep, maxSnapshotAgeMs)
+        .commit();
+    return this;
+  }
+
+  @Override
+  public ManageSnapshots createTag(String name, long snapshotId) {
+    return createTag(name, snapshotId, null);
+  }
+
+  @Override
+  public ManageSnapshots createTag(String name, long snapshotId, Long maxRefAgeMs) {
+    transaction.manageSnapshotReferenceOperation().createTag(name, snapshotId, maxRefAgeMs).commit();
```

Comment:
Can you update this so that the last `manageSnapshotReferenceOperation` is reused? That way we don't create a ton of unnecessary operations in the transaction.

Reply:
Makes sense, will update.